util/comparator.cc

```diff
@@ -216,6 +218,70 @@ class ReverseBytewiseComparatorImpl : public BytewiseComparatorImpl {
     return -a.compare(b);
   }
 };
+
+// EXPERIMENTAL
+// Comparator with 64-bit integer timestamp.
+// We did not performance test this yet.
+template <typename T>
+class ComparatorWithU64TsImpl : public Comparator {
+  static_assert(std::is_base_of<Comparator, T>::value, "T must be a inherited type of comparator");
+ public:
+  ComparatorWithU64TsImpl()
+      : Comparator(/*ts_sz=*/sizeof(uint64_t)),
+        cmp_without_ts_(BytewiseComparator()) {
```

Comment:
Need to change constructor to pass a 
ComparatorWithU64TsImpl(T *). In the constructor I will cast and assign to Comparator*